### PR TITLE
refactor core code to use `axios` instead of `request`

### PR DESCRIPTION
**Pull request by dpopp07**
This PR migrates the Node SDK to use [`axios`](https://github.com/axios/axios) instead of [`request`](https://github.com/request/request) for making network requests. This involved some minor modifications to the code:
- Constructing a config object formatted for `axios` with the request data
- Building FormData parameters internally, since this is not handled by `axios`
- Adding `responseType` to several SDK methods that return Streams (this should be generated in the future)
- Refactoring tests that relied on `request` returning a Stream, as `axios` does not return streams
- I also made slight code changes to improve readability

Looking for feedback now but there is still work to do.

What is still left to do?
- [x] Add code to perform the GZIPing that `request` did
- [x] Go through and make sure `responseType` is set everywhere it needs to be
- [x] Add code for setting `rejectUnauthorized` to `false` for ICP compatibility
- [x] Test with ICP
- [x] Determine an interface for the response
- [x] Determine an interface / format for errors as the error formatting method is currently not being used in the new code
- [x] Document any breaking changes in the UPGRADE file

### CODE REVIEW

File: lib/requestwrapper.ts [hunk 1]
```diff
@@ -74,14 +76,14 @@ export function formatErrorIfExists(cb: Function): request.RequestCallback {
 
     // for api-key services
     if (response.statusMessage === 'invalid-api-key') {
-      const error = {
+      const err = {
         error: response.statusMessage,
         code: response.statusMessage === 'invalid-api-key' ? 401 : 400,
```
Comment (germanattanasio) [inline lib/requestwrapper.ts]: this was needed for alchemy we can probably remove it now

File: lib/requestwrapper.ts [hunk 2]
```diff
@@ -179,59 +156,111 @@ export function sendRequest(parameters, _callback) {
...
+
+  if (formData) {
+    data = multipartForm;
+    // form-data generates headers that MUST be included or the request will fail
+    headers = extend(true, {}, headers, multipartForm.getHeaders());
   }
 
-  // Compression support
-  options.gzip = true;
+  const axiosObject = {
```
Comment (germanattanasio) [inline lib/requestwrapper.ts]: I would call it `requestParams`. I don't think we need `axios` to be part of the name

File: package.json [hunk 3]
```diff
@@ -80,10 +80,12 @@
     "@types/node": "~10.3.5",
     "@types/request": "~2.47.1",
     "async": "~2.6.1",
+    "axios": "^0.18.0",
```
Comment (germanattanasio) [inline package.json]: Can you remove `request`?

File: lib/requestwrapper.ts [hunk 4]
```diff
@@ -41,93 +44,92 @@ function parsePath(path: string, params: Object): string {
...
+      error['transaction-id'] = axiosError.headers['transaction-id'];
     }
 
-    // ensure a more descriptive error message
-    if (error && (error.code === 401 || error.code === 403)) {
-      error.body = error.message;
-      error.message = 'Unauthorized: Access is denied due to invalid credentials.';
+    // other services use x-global-transaction-id
+    if (axiosError.headers['x-global-transaction-id']) {
+      error['x-global-transaction-id'] = axiosError.headers['x-global-transaction-id'];
```
[the remark this answers is not in the document]
Comment (dpopp07) [reply lib/requestwrapper.ts]: Yep, I can do that. Didn't know if that would make it too crowded but I like giving them that info

**Comment by dpopp07** [comment]
@germanattanasio Work is complete and tests are updated. PR is ready for review

**Pull request merged**